usdoj-crt/beta-ada
# Optimize Images

This is a followup to introducing the new Design Standards images for https://github.com/usdoj-crt/beta-ada/pull/487

In that PR, I noticed our image assets were relatively unoptimized, so I ran a utility to make some quick improvements. Further improvements could be made wherever we don't mind sacrificing image resolution or marginal quality (for example, on the topics page).

## Why does this matter?

Pages with large images (such as the topics page) can take a long time to load, especially over 3G or satellite internet connections. These images also take valuable bandwidth - both for users with monthly bandwidth limits (mobile data and some desktop users), and ada.gov servers.

*Note: this GIF taken _after_ compression*

![slow](https://user-images.githubusercontent.com/15126660/215882243-792282c4-a5c1-4b41-9b1a-8234ae28cecd.gif)

## How to review this:

(I also did this for all images myself, but would appreciate the sanity check)

For each image attached:

- Confirm that the new image is smaller than the old one. This should be red and show a negative number: <img width="590" alt="image" src="https://user-images.githubusercontent.com/15126660/215879688-148ceb4e-49f1-4ebf-8b01-191df382d56b.png">
- Confirm that the modified image looks the same as the old one (there's no loss in color, quality, etc). You can click "Swipe" to get a larger view: <img width="893" alt="image" src="https://user-images.githubusercontent.com/15126660/215880138-8963a184-b0bf-4bc1-b0e2-0e174464abd9.png">
- If the image shows "unable to render rich display", click "display the source diff" on the right to re-render it: <img width="1479" alt="image" src="https://user-images.githubusercontent.com/15126660/215880484-5809d6a7-5508-4a83-b631-bdd3aaf03897.png">

## What does this change?

- 🌎 Images can often be compressed to take up less space without
  looking worse. Compressed images:
  - Download faster, serving users with slower connection speeds better
  - Take up less bandwith to deliver
- ⛔ A bunch of the ADA images showed significant size decrease with effectively lossless optimization.
- ✅ Running a compression utility (`jpegoptim` and `optipng`) on our ADA images resulted in
  significant gains in a few places

In total, the `_assets/images` was reduced from 38MB to 14MB:

Before: 

```shell
└─ $ du -h _assets/images
29M     _assets/images/landing
5.4M    _assets/images/project-images
2.6M    _assets/images/design-standards
38M     _assets/images
```

After:

```
└─ $ du -h _assets/images
5.1M    _assets/images/landing
4.5M    _assets/images/project-images
2.6M    _assets/images/design-standards
14M     _assets/images
```
